dali/base/dadfs.cpp

```diff
@@ -12637,7 +12638,11 @@ SecAccessFlags CDistributedFileDirectory::getFDescPermissions(IFileDescriptor *f
                 CDfsLogicalFileName dlfn;
                 dlfn.setExternal(rfn.queryEndpoint(),localpath.str());
                 SecAccessFlags perm = getDLFNPermissions(dlfn,user,auditflags);
-                if (perm < retPerms) {
+                if (!accessSet) {
+                    retPerms = perm;
+                    accessSet = true;
+                }
+                else if (perm < retPerms) {
                     retPerms = perm;
                     if (retPerms == SecAccess_None)
```

Comment:
this shortcircuit code should apply to the !accessSet branch too, i.e. should move outside (after) the condition.